### DISCUSSION TITLE
Create a wallet through PrivateKey, and get the wallet address?

### DISCUSSION THREAD

**Discussion by Olive-l** [trustwallet/wallet-core#2968] [Q&A] [answered]
**Is your feature request related to a problem? Please describe.**
I am an APP developer. I wanna create a wallet through private key (instead mnemonic) in my app. By the way, the private key which is exported from metamask.

The code is:
`PrivateKey.init(data)`
and it's worked.

After that, I can get the public key. But what I want is wallet address. Is there any possible to do that?

**Comment by Milerius** [comment, marked as answer]
Hello @Olive-l , which programming language are u using?

**Reply by Olive-l** [reply]
swift in iOS platform

**Comment by Olive-l** [comment]
@Milerius I have fixed this, many thanks all the same.
